backend/routes/resetCourts.js

```diff
@@ -0,0 +1,96 @@
+const express = require("express");
+const router = express.Router();
+const admin = require("firebase-admin");
+const bcrypt = require("bcrypt");
+
+function bcryptCompare(plaintext, hashed) {
+  return new Promise(function (resolve, reject) {
+    bcrypt.compare(plaintext, hashed, function (err, res) {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(res);
+      }
+    });
+  });
+}
+
+async function resetActivePlayers(location, batch) {
+  const active_players = await location
+    .collection("activePlayers")
+    .listDocuments();
+
+  for (var j = 0; j < active_players.length; j++) {
+    const player_ref = await location
+      .collection("activePlayers")
+      .doc(active_players[j].id);
+    const player = await (await player_ref.get()).data();
+
+    const court_number = player.courtNumber;
+    const new_name = "Empty" + court_number.toString();
+    player.playerWaiting = false;
+    player.firebaseUID = new_name;
+    player.nickname = new_name;
+    const updateObj = {
+      playerWaiting: player.playerWaiting,
+      firebaseUID: player.firebaseUID,
+      nickname: player.nickname,
+    };
+    batch.update(player_ref, updateObj);
+  }
+}
+
+async function deleteQueuePlayers(location, batch) {
+  const queue_players = await location
+    .collection("queuePlayers")
+    .listDocuments();
+
+  for (var j = 0; j < queue_players.length; j++) {
+    const player_ref = await location
+      .collection("queuePlayers")
+      .doc(queue_players[j].id);

```

Comment:
I believe this may causing an additional read each iteration since the "listDocuments" above the loop already gets the document references.

Can we try something like:

queue_players.forEach((docRef) => {
    batch.delete(docRef);  // As reference already retrieved
  });